nautobot/dcim/api/serializers.py

```diff
@@ -545,12 +547,22 @@ def get_field_names(self, declared_fields, info):
         fields = list(super().get_field_names(declared_fields, info))
         self.extend_field_names(fields, "parent_bay")
         self.extend_field_names(fields, "config_context", opt_in_only=True)
+        if "cluster" not in fields:
+            fields.append("cluster")
         return fields
 
     @extend_schema_field(serializers.DictField)
     def get_config_context(self, obj):
         return obj.get_config_context()
 
+    @extend_schema_field(serializers.DictField)
+    def get_cluster(self, obj):
+        """Return the first cluster for backward compatibility"""
+        try:
+            return obj.cluster.pk if obj.cluster else None
+        except obj.clusters.model.MultipleObjectsReturned:
+            return obj.clusters.first().pk if obj.clusters.exists() else None
```

Comment:
The pattern for VLAN.location and Prefix.location is to raise a helpful exception if there are multiple values, rather than silently discarding the additional values.

Reply:
the `Device.cluster` attr raises `MultipleObjectsReturned` in the same way as `VLAN.location`

I think this schema field is more junk leftover from when I was iterating; removed.